precompiles/bank/Bank.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+address constant BANK_PRECOMPILE_ADDRESS = 0x0000000000000000000000000000000000001001;
+
+IBank constant BANK_CONTRACT = IBank(
+    BANK_PRECOMPILE_ADDRESS
+);
+
+interface IBank {
+    // Transactions
+    function send(
+        address fromAddress,
+        address toAddress,
+        string memory denom,
+        uint256 amount
+    ) external returns (bool success);
+
+    function sendFromCaller(
+        address toAddress,
+        string memory denom,
+        uint256 amount
+    ) external returns (bool success);
+
+    function sendFromOrigin(
+        address toAddress,
+        string memory denom,
+        uint256 amount
+    ) external returns (bool success);
```

Comment:
So send allows you to send from any address, sendFromCaller only allows you to send from msg.sender and sendFromOrigin allows you to send from tx.origin?

Reply:
yeah, and `send` is only callable by whitelisted code hashes